rfcs/0008-microtvm-project-api.md

```diff
@@ -0,0 +1,518 @@
+3. `tvm.micro.Transport`: used to communicate with on-device microTVM RPC server
+
+Thus far, implementations of these interfaces have been made for Zephyr, mBED OS, and for
+simulated hardware using a POSIX subprocess. The latter two interfaces have proven to be a
+relatively good fit; however, `tvm.micro.Compiler` is difficult to implement because it attempts
+to replicate a platform's build system in TVM. TVM does not want to incorporate platform-specific
+build logic into its codebase.
+
+This proposal unifies these three interfaces to form a "Project-level" interface, recognizing that
+it's typical to interact with unconventional platforms and their build systems at this level. It
+simplifies the `Compiler` interaction into a project-level Build, and adds an explicit
+`generate_project` method to the interface. These changes remove the need to build components
+and drive the link process from TVM.
+
+As a goal, this proposal aims to allow for the same use cases as are currently supported with these
+improvements:
+
+1. Integrating more naturally with build systems typical of embedded platforms.
+2. Allowing TVM to automatically generate projects  platforms to define automated scripts to build projects
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+TVM can interact with platform SDKs via its **Project API**. Such SDKs are common when working with
+non-traditional OS platforms, such as those commonly used in embedded systems (e.g. Arduino, Zephyr,
+iOS). Given a platform-specific implementation of this Project API, TVM can:
+1. Generate projects that integrate implemented TVM models with generic platform runtime componeents
+2. Build those generated projects
+3. Program attached hardware
+4. Drive remote model execution via the TVM RPC Server interface.
+
+This last capability means that TVM can drive autotuning, remotely perform model inference, and debug
+models on non-traditional OS such as Arduino, Zephyr, and mobile platforms such as iOS and Android.
+
+To provide support for a platform, a **template project** is first defined. Template projects are
+expected to exist entirely inside a directory and are identified to TVM by the path to the directory
+locally. Template projects may live either inside the TVM repository (when they can be included in the
+TVM CI) or in other version control repositories. The template project contains at minium an
+implementation of the **Project API** inside an executable program known as the
+**TVM Project API Server**.
+
+To begin working with a particular platform's Project API implementation, the user supplies TVM with
+the path to the top-level directory. TVM launches an instance the Project API Server (found at a
+standard location in that directory). TVM communicates with the Project API Server using JSON-RPC
+over standard OS pipes.
+
+TVM supplies generated code to the Project API Server using [Model Library Format](0001-model-library-format.md).
```

Comment:
This link is broken. Is there an RFC for model library format on the way?

Reply:
oh thanks for catching--it's actually just docs in https://github.com/apache/tvm/pull/8270 which should land soon

Reply:
sorry, but this needs to change again since the PR is merged :(